racket/src/cs/rumble/hash.ss

```diff
@@ -257,31 +257,36 @@
 (define hash-ref
   (case-lambda
     [(ht k)
-     (let ([v (hash-ref ht k none)])
+     (let ([v (hash-ref/none ht k)])
        (if (eq? v none)
            (raise-arguments-error
             'hash-ref
             "no value found for key"
             "key" k)
            v))]
-    [(ht k fail)
-     (cond
-      [(mutable-hash? ht)
-       (lock-acquire (mutable-hash-lock ht))
-       (let ([v (hashtable-ref (mutable-hash-ht ht) k none)])
-         (lock-release (mutable-hash-lock ht))
-         (if (eq? v none)
-             ($fail fail)
-             v))]
-      [(intmap? ht) (intmap-ref ht k fail)]
-      [(weak-equal-hash? ht) (weak-hash-ref ht k fail)]
-      [(and (impersonator? ht)
-            (authentic-hash? (impersonator-val ht)))
-       (let ([v (impersonate-hash-ref ht k)])
-         (if (eq? v none)
-             ($fail fail)
-             v))]
-      [else (raise-argument-error 'hash-ref "hash?" ht)])]))
+    [(ht k default)
+     (let ([v (hash-ref/none ht k)])
+       (if (eq? v none)
+           (fail-hash-ref default)
+           v))]))
+
+(define (hash-ref/none ht k)
```

Comment:
i really wanted to make this `hash-ref/default` and have `hash-ref` pass `none` into it, but I ran into the fact that `impersonate-hash-ref` doesn't have any parameter for a default, and I was a bit scared to add one.